### ISSUE TITLE
Enhancement: open p:file-copy p:file-delete p:file-move to sequence of anyURIs

### ISSUE THREAD

**Issue by Arithmeticus** [xproc/3.0-steps#467]
Enhancement suggestion: open p:file-copy, -delete, and -move to apply to multiple input items, either by changing option href to permit a sequence of anyURIs or by providing a new option to filter via glob-like syntax. I realize you can embed these commands in iterative constructions, but I think my suggestion could help create cleaner, more maintainable code.

**Comment by xml-project** [comment]
@Arithmeticus this issue is currently under investigation. 
Our first option would be to leave the steps as they are, but supply a library of step written in XProc which take a path option, an include-filter and an exclude-filter (both regex). This step would simply be a p:directory-list with the given values and then an iteration over all results performing the copy, delete or move action.
However if that turns out to be to slow or to inconvinient to user, we will are open to add additional steps (e.g. p:file-group-copy, p:file-group-delete and file-group-move) with the requested functionality.

**Comment by Arithmeticus** [comment]
@xml-project I agree, and add that in many file operations, users would expect to access other parameters, such as specifying conditions under which files should be overwritten (date, file size). Also, I think most users expect to use globlike syntax, not regular expressions ('though I prefer the latter). I have hammered out some prototypes along the lines mentioned under the branch xproc-test of this repo:
https://github.com/Arithmeticus/XML-Pantry/tree/xproc-test  

That includes a function `ladder:glob-to-regex()` that could be useful in other contexts.

**Comment by xml-project** [comment]
@Arithmeticus Yes, that makes sense to me. Thanks. Could you check your supplied link please. It gives me a 404.

**Comment by Arithmeticus** [comment]
Oops! https://github.com/Arithmeticus/XML-Pantry/tree/xproc-test

**Comment by xatapult** [comment]
Will be moved to vnext as of editors call 2020-10-22
